lib/datadog/open_feature/evaluation_engine.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require_relative 'ext'
+require_relative 'binding'
+require_relative 'noop_evaluator'
+
+module Datadog
+  module OpenFeature
+    # This class performs the evaluation of the feature flag
+    class EvaluationEngine
+      attr_accessor :configuration
+      attr_reader :reporter
+
+      ALLOWED_TYPES = %i[boolean string number float integer object].freeze
```

Comment:
This seems to be inconsistent with this change:
https://github.com/DataDog/dd-trace-rb/pull/5024/files#diff-5c8c3a7c7fe1084d00c909ef1ff163c31dd21cf02dedc4ed05ee1eef2c39158cR12